vignettes/relase-process.qmd

```diff
@@ -0,0 +1,198 @@
+
+### Releases
+
+Releases are a concept that is specific for GitHub. Under the hood, releases are
+created from commits or tags. When you use the GitHub web interface, you can
+choose to have a tag automatically created from a release. 
+
+Once the release is created on GitHub, then the package will be available on
+the R-Universe in about an hour or less. 
+
+It is good practice to create a tag to release from so that we have a way to
+track the releases even if we eventually move away from GitHub. Even better
+practice is to create _signed_ tags, which adds a cryptographic signature to
+the tag that allows anyone to verify that it came from your computer and not
+someone pretending to be you.
+
+#### Signed Tags
+
+Zhian likes to create tags via the command line because he has set up his git
+configuration to use [a gpg signature](https://www.garrickadenbuie.com/blog/signed-verified-git-commits-keybase-rstudio/#fnref3)
+so the tags and the releases are both verified. Recently, Git and GitHub added
+support for [creating signatures via an SSH key](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#ssh-commit-signature-verification), which is _a lot_ more approchable than GPG.
+
+Tagging and Release can be achieved in a single step with the [github
+cli](https://cli.github.com/) with the command `gh release create X.Y.Z` for
+the version number
+
+```bash
+gh release create 3.3.3
+# ? Title (optional) sandpaper 3.3.3
+# ? Release notes  [Use arrows to move, type to filter]
+#   Write my own
+# > Write using generated notes as template
+#   Leave blank
+```
+
+Selecting "Write using generated notes as a template" opens an editor and
+populates it with the pull requests that have been accepted since the last
+release.
+
+
+## Release Timeline Example
+
+In the early days of most software, the release process is ad-hoc because the
+number of users is low, but with an entire community is using it, it is
+**important to set expectations and properly communicate changes to the users**.
+For [patch relases](#patch), these releases should continue to happen ad-hoc
+because, by definition, these releases only fix bugs and do not add new
+features.
+
+The release timelines are based on one fundamental assumption: 
+
+1. a M--F workweek (i.e. NEVER release on a Friday unless you like working on
+   the weekend)
+
+### Minor Feature Updates
+
+For releases that add new non-breaking features which bump the [minor
+version](#minor), it is important to alert the community a couple of days in
+advance of the release so they have a chance to give feedback before the
+release. Thus, I propose the following timeline:
+
+ 1. **Friday (drafting)**: Members of Maintainer Group draft
+    communications describing new features
+ 2. **Monday (communications)**: Communications describing new features sent to
+    relevant communities 
+ 3. **Wednesday (release)**: New version of package(s) released
```

Comment:
Overall this schedule is not ideal for the hubverse as many hubs collect submissions on a wednesday so not the best day for releasing new versions of packages. Also given not all of us are full time/work every day of the week, I wonder if this is a bit too restrictive for us atm.

Reply:
I had a feeling that was the case; I definitely don't want to put too many restrictions on this so I'll go ahead and strike it out.

Reply:
It has been struck out